File: EmptyProject/WorkflowsTrainingTwo/WorkflowsTrainingCPlus/Parsing/Expressions/Nodes/Expressions/NumberLiteralNode.cpp
#include "NumberLiteralNode.h"
#include <iostream>

namespace WorkflowsTraining::Parsing::Expressions::Nodes::Expressions {
    NumberLiteralNode::NumberLiteralNode(int value)
        : value_(value) {
    }

    std::variant<bool, int, double, std::string> NumberLiteralNode::Resolve() {
        std::cout << "NumberLiteral: Resolving to " << value_ << std::endl;
        return value_;
    }
}

File: EmptyProject/WorkflowsTrainingTwo/WorkflowsTraining/Parsing/Expressions/Nodes/Expressions/NumberLiteralNode.cs
using WorkflowsTraining.Parsing.Expressions.Interfaces;

namespace WorkflowsTraining.Parsing.Expressions.Nodes.Expressions;

public class NumberLiteral(int value) : IExpressionNode
{
    private int Value { get; } = value;

    public object Resolve()
    {
        return Value;
    }
}